packages/nodes-base/nodes/Form/FormTrigger.node.ts

```diff
@@ -0,0 +1,278 @@
+								displayOptions: {
+									show: {
+										fieldType: ['dropdown'],
+									},
+								},
+							},
+							{
+								displayName: 'Required Field',
+								name: 'requiredField',
+								type: 'boolean',
+								default: false,
+								description:
+									'Whether to require the user to enter a value for this field before submitting the form',
+							},
+						],
+					},
+				],
+			},
+			{
+				displayName: 'Respond When',
+				name: 'responseMode',
+				type: 'options',
+				options: [
+					{
+						name: 'Form Is Submitted',
+						value: 'onReceived',
+						description: 'As soon as this node receives the form submission',
+					},
+					{
+						name: 'Workflow Finishes',
+						value: 'lastNode',
+						description: 'When the last node of the workflow is executed',
+					},
+				],
+				default: 'onReceived',
+				description: 'When to respond to the form submission',
+			},
+			{
+				displayName: 'Options',
+				name: 'options',
+				type: 'collection',
+				placeholder: 'Add Option',
+				default: {},
+				options: [
+					{
+						displayName: 'Form Submitted Text',
+						name: 'formSubmittedText',
+						description: 'The text displayed to users after they filled the form',
+						type: 'string',
+						default: 'Your response has been recorded',
+					},
+				],
+			},
+		],
+	};
+
+	async webhook(this: IWebhookFunctions): Promise<IWebhookResponseData> {
+		const webhookName = this.getWebhookName();
+		const mode = this.getMode() === 'manual' ? 'test' : 'production';
+		const formFields = this.getNodeParameter('formFields.values', []) as FormField[];
+
+		//Show the form on GET request
+		if (webhookName === 'setup') {
+			const formTitle = this.getNodeParameter('formTitle', '') as string;
+			const formDescription = this.getNodeParameter('formDescription', '') as string;
+			const instanceId = await this.getInstanceId();
+			const { formSubmittedText } = this.getNodeParameter('options', {}) as IDataObject;
+
+			const page = createPage(
+				formTitle,
+				formDescription,
+				formSubmittedText as string,
+				formFields,
+				mode === 'test',
+				instanceId,
+			);
+
+			const res = this.getResponseObject();
+			res.status(200).send(page).end();
+			return {
+				noWebhookResponse: true,
+			};
+		}
+
+		const bodyData = (this.getBodyData().data as IDataObject) ?? {};
+
+		const returnData: IDataObject = {};
+		for (const [index, field] of formFields.entries()) {
+			const key = `field-${index}`;
+			let value = bodyData[key] ?? null;
+
+			if (value === null) returnData[field.fieldLabel] = null;
+
+			if (field.fieldType === 'number') {
+				value = Number(value);
+			}
+			if (field.fieldType === 'text') {
+				value = String(value).trim();
+			}
+			if (field.multiselect && typeof value === 'string') {
+				value = jsonParse(value);
+			}
+
+			returnData[field.fieldLabel] = value;
+		}
+		returnData.submittedAt = bodyData.submittedAt;
```

Comment:
should not this be set serverside? rather than accept it from the FE.. or at least we should validate the value